### PR TITLE
fix: Tickets list is not updated after creating new ticket #1446

### LINKED ISSUE #1446: Tickets list is not updated after creating new ticket
**Describe the bug**
Whenever we create a new ticket, and return to the screen containing the list of the tickets, then we need to manually refresh the contents to see the newly added ticket. 

**To Reproduce**
Go to the Tickets section, create a new ticket and return to the tickets screen. The new ticket will nnot be present there. 

**Expected behavior**
The list should be updated. 

**Logs**
NA

**Screenshots**
ezgif com-video-to-gif 1

**Smartphone Info:**
|Device|         Redmi Note 4
|Android Version|  Nougat 7.1.1

**Additional context**

**Would you like to work on the issue?**

- [X] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by tavishjain**
Fixes #1446 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream development branch.

Changes: Whenever the tickets fragment is visited, the list of tickets is refreshed, and then displayed

Screenshots for the change:
ezgif com-video-to-gif

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsFragment.java [hunk 1]
```diff
@@ -81,6 +81,7 @@ public void onStart() {
         setupRecyclerView();
         setupRefreshListener();
         getPresenter().attach(eventId, this);
+        getPresenter().loadTickets(true);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsFragment.java]: It should be working automatically. Test why the Observable is not working correctly

Comment (tavishjain) [reply app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsFragment.java]: Sure

**Comment by ShridharGoel** [comment]
This is the problem I was talking about in https://github.com/fossasia/open-event-server/issues/5521

**Comment by iamareebjamal** [comment]
The dev server was reset so yes you needed to create a new account. Reminder: Dev server is regularly reset

**Comment by tavishjain** [comment]
@iamareebjamal but then why am I not reciving the confirmation mail again ?

**Comment by iamareebjamal** [comment]
Dev server is not configured correctly to send the mails. Need to check it and I don't have access

**Comment by tavishjain** [comment]
So how should I proceed for ow ? @iamareebjamal

**Comment by iamareebjamal** [comment]
You can either set up the server locally or wait for a few days for dev server to be configured correctly

**Comment by tavishjain** [comment]
@iamareebjamal Please review the changes

**Comment by tavishjain** [comment]
@iamareebjamal can you please explain what you are trying to say ?

File: app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsPresenter.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public TicketsPresenter(TicketRepository ticketRepository, DatabaseChangeListene
 
     @Override
     public void start() {
-        loadTickets(false);
+        loadTickets(true);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsPresenter.java]: Why changing to true?

Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsPresenter.java]: When a ticket is created, it is inserted in the db, so ideally this should work without fetching all the tickets from server

Comment (tavishjain) [reply app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsPresenter.java]: @iamareebjamal I checked out the code for a pretty while and found that the data being pulled on in the fragment is being accessed by the Presenter and this reload parameter has to be true is order to get refreshed. Even though the data is stored in a database, this parameter refreshes the items whenever the user returns or moves to the tickets fragment.

Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/ticket/list/TicketsPresenter.java]: true is passed when we want to fetch from internet

**Pull request closed without merging**